model/task/task.go

```diff
@@ -3215,6 +3205,16 @@ func (t *Task) PopulateTestResults(ctx context.Context) error {
 	return nil
 }
 
+// AppendTestResults returns the task's test results filtered, sorted, and
+// paginated as specified by the optional filter options.
+func (t *Task) AppendTestResults(ctx context.Context, testResults []testresult.TestResult) error {
```

Comment:
I don't think we want to have any write functionality directly on the task struct because that should happen on the agent.

Reply:
Removed